piicatcher/dbinfo.py

```diff
@@ -82,10 +87,55 @@ def get_sample_query(
 
 
 class Redshift(Postgres):
-    # Redshift doesn't have BERNOULLI distribution. RANDOM() is the closest we can get to sample data from tables.
     _sample_query_template = "SELECT {column_list} FROM {schema_name}.{table_name} ORDER BY RANDOM() LIMIT {num_rows}"
 
 
+class BigQuery:
```

Comment:
Does this class need to derive from `DbInfo` as well?

```suggestion
class BigQuery(DbInfo):
```

Reply:
originally I thought it was needed but it is unable to reuse the template query and the implementation of the methods in DbInfo. I'm not sure if it needs to be derived from DbInfo in that case.

Reply:
That's probably because these connector classes are not really used as classes in the OO sense, but more as collections of independent functions. As per my previous comment, I would consider refactoring that so that each concrete DbInfo instance becomes an actual object that can potentially hold state like a BigQuery project ID.

I.e. I would change `get_dbinfo` into something like this:

```python
def get_dbinfo(source_type: str, *args, **kwargs) -> DbInfo:
    if source_type == "sqlite":
        return Sqlite(*args, **kwargs)
    ...
    elif source_type == "bigquery":
        return BigQuery(*args, **kwargs)
   raise AttributeError
```

Then you also don't need special treatment for BigQuery in `generators.py` any more:

```python
    if source.source_type == "bigquery":
        count = dbinfo.get_count_query(schema.name, table.name, source.project_id)
    else:
        count = dbinfo.get_count_query(schema.name, table.name)
```

simply becomes

```python
   count = dbinfo.get_count_query(schema.name, table.name)
```

since at that point `dbinfo` is an instance of the `BigQuery` class and the project ID was already passed when the object was initialized and can be used in all the instance methods.